src/otx/cli/utils/jsonargparse.py

```diff
@@ -0,0 +1,140 @@
+    """
+    _dict_type = False
+    if isinstance(value, dict):
+        # Dict -> Nested Namespace for overriding
+        _dict_type = True
+        value = dict_to_namespace(value)
+    if not isinstance(value, (Namespace, dict)):
+        if not key:
+            msg = "Key is required if value not a Namespace."
+            raise KeyError(msg)
+        if not only_unset or key not in self:
+            if key not in self or value is not None:
+                if isinstance(value, str) and (value.isnumeric() or value in ("True", "False")):
+                    value = ast.literal_eval(value)
+                self[key] = value
+            elif value is None:
+                del self[key]
+    else:
+        prefix = key + "." if key else ""
+        for _key, val in value.items():
+            if not only_unset or prefix + _key not in self:
+                self.update(val, prefix + _key)
+    if _dict_type and key is not None:
+        # Dict or Namespace -> Dict
+        self[key] = dict_to_namespace(self[key]).as_dict()
+    return self
+
+
+# To provide overriding of the Config file
+def apply_config(self: ActionConfigFile, parser: ArgumentParser, cfg: Namespace, dest: str, value: str) -> None:  # noqa: ARG001
+    """Applies the configuration to the parser.
+
+    Args:
+        parser: The parser object.
+        cfg: The configuration object.
+        dest: The destination attribute.
+        value: The value to be applied.
+
+    Returns:
+        None
+    """
+    from jsonargparse._actions import _ActionSubCommands, previous_config_context
+    from jsonargparse._link_arguments import skip_apply_links
+    from jsonargparse._loaders_dumpers import get_loader_exceptions, load_value
+    from jsonargparse._optionals import get_config_read_mode
+
+    with _ActionSubCommands.not_single_subcommand(), previous_config_context(cfg), skip_apply_links():
+        kwargs = {"env": False, "defaults": False, "_skip_check": True, "_fail_no_subcommand": False}
+        try:
+            cfg_path: Path | None = Path(value, mode=get_config_read_mode())
+        except TypeError:
+            try:
+                if isinstance(load_value(value), str):
+                    raise
+                cfg_path = None
+                cfg_file = parser.parse_string(value, **kwargs)
+            except (TypeError, *get_loader_exceptions()) as ex_str:
+                msg = f'Parser key "{dest}": {ex_str}'
+                raise TypeError(msg) from ex_str
+        else:
+            cfg_file = parser.parse_path(value, **kwargs)
+        cfg_merged = parser.merge_config(cfg_file, cfg)
+        cfg.__dict__.update(cfg_merged.__dict__)
+        overrides = cfg.__dict__.pop("overrides", None)
+        if overrides is not None:
+            cfg.__dict__.update(overrides)
+        if cfg.get(dest) is None:
+            cfg[dest] = []
+        cfg[dest].append(cfg_path)
+
+
+Namespace.update = update
+ActionConfigFile.apply_config = apply_config
```

Comment:
Can we add a kind of contextmanager here rather than silently monkey-patching the upstream library?

Reply:
Thanks - Done https://github.com/openvinotoolkit/training_extensions/pull/2805/commits/768951e218e5b4b56ce29516c3c2dd73b44e5615